worlds/osrs/Rules.py

```diff
@@ -0,0 +1,333 @@
+
+
+def get_magic_skill_rule(level, player, options) -> CollectionRule:
+    if options.max_magic_level < level:
+        return lambda state: False
+
+    return lambda state: state.can_reach_region(RegionNames.Mind_Runes, player)
+
+
+def get_firemaking_skill_rule(level, player, options) -> CollectionRule:
+    if options.max_firemaking_level < level:
+        return lambda state: False
+    if not options.brutal_grinds:
+        if level >= 30:
+            can_chop_willows = get_woodcutting_skill_rule(30, player, options)
+            return lambda state: state.can_reach_region(RegionNames.Willow_Tree, player) and can_chop_willows(state)
+        if level >= 15:
+            can_chop_oaks = get_woodcutting_skill_rule(15, player, options)
+            return lambda state: state.can_reach_region(RegionNames.Oak_Tree, player) and can_chop_oaks(state)
+    # If brutal grinds are on, or if the level is less than 15, you can train it.
+    return lambda state: True
+
+
+def get_skill_rule(skill, level, player, options) -> CollectionRule:
+        if skill.lower() == "fishing":
+            return get_fishing_skill_rule(level, player, options)
+        if skill.lower() == "mining":
+            return get_mining_skill_rule(level, player, options)
+        if skill.lower() == "woodcutting":
+            return get_woodcutting_skill_rule(level, player, options)
+        if skill.lower() == "smithing":
+            return get_smithing_skill_rule(level, player, options)
+        if skill.lower() == "crafting":
+            return get_crafting_skill_rule(level, player, options)
+        if skill.lower() == "cooking":
+            return get_cooking_skill_rule(level, player, options)
+        if skill.lower() == "runecraft":
+            return get_runecraft_skill_rule(level, player, options)
+        if skill.lower() == "magic":
+            return get_magic_skill_rule(level, player, options)
+        if skill.lower() == "firemaking":
+            return get_firemaking_skill_rule(level, player, options)
+
+        return lambda state: True
+
+
+def generate_special_rules_for(entrance, region_row, outbound_region_name, player, options):
+    if outbound_region_name == RegionNames.Cooks_Guild:
+        add_rule(entrance, get_cooking_skill_rule(32, player, options))
+    if outbound_region_name == RegionNames.Crafting_Guild:
+        add_rule(entrance, get_crafting_skill_rule(40, player, options))
+    if outbound_region_name == RegionNames.Corsair_Cove:
+        # Need to be able to start Corsair Curse in addition to having the item
+        add_rule(entrance, lambda state: state.can_reach(RegionNames.Falador_Farm, "Region", player))
+    if outbound_region_name == "Camdozaal*":
+        add_rule(entrance, lambda state: state.has(ItemNames.QP_Below_Ice_Mountain, player))
+    if region_row.name == "Dwarven Mountain Pass" and outbound_region_name == "Anvil*":
+        add_rule(entrance, lambda state: state.has(ItemNames.QP_Dorics_Quest, player))
```

Comment:
I assume you can do this instead? (And a few other places later on in this function)
```suggestion
    if outbound_region_name == RegionNames.Cooks_Guild:
        add_rule(entrance, get_cooking_skill_rule(32, player, options))
    elif outbound_region_name == RegionNames.Crafting_Guild:
        add_rule(entrance, get_crafting_skill_rule(40, player, options))
    elif outbound_region_name == RegionNames.Corsair_Cove:
        # Need to be able to start Corsair Curse in addition to having the item
        add_rule(entrance, lambda state: state.can_reach(RegionNames.Falador_Farm, "Region", player))
    elif outbound_region_name == "Camdozaal*":
        add_rule(entrance, lambda state: state.has(ItemNames.QP_Below_Ice_Mountain, player))
    elif region_row.name == "Dwarven Mountain Pass" and outbound_region_name == "Anvil*":
        add_rule(entrance, lambda state: state.has(ItemNames.QP_Dorics_Quest, player))
```